cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/co/cask/cdap/etl/common/PipelineRegisterer.java

```diff
@@ -159,10 +162,28 @@ public Pipeline registerPlugins(ETLConfig config, Class errorDatasetType, Datase
     return new Pipeline(sourcePluginId, sinksInfo, transformInfos);
   }
 
+  private void validateStageNames(ETLStage sourceConfig, List<ETLStage> transforms, List<ETLStage> sinks) {
+    Set<String> uniquieStageNames = new HashSet<>();
+    uniquieStageNames.add(sourceConfig.getName());
+    validateUniqueStageName(uniquieStageNames, transforms);
+    validateUniqueStageName(uniquieStageNames, sinks);
+  }
+
+  private void validateUniqueStageName(Set<String> uniqueStageNames, List<ETLStage> stages) {
+    for (ETLStage stage : stages) {
+      if (uniqueStageNames.contains(stage.getName())) {
```

Comment:
Set.add() returns a boolean, so you could use that instead of checking then adding.